src/v5_hal/test/pathing/test_PathPoint.cpp

```diff
@@ -0,0 +1,118 @@
+#include <gtest/gtest.h>
+#include "pathing/PathPoint.h"
+
+class xChangeConstantVelModule : public ::testing::Test {
+    protected:
+        PathPoint* firstPoint;
+        PathPoint* secondPoint;
+
+        virtual void SetUp() {
+            // Creates two points that only have a positive change in x with constant velocity
+            firstPoint = new PathPoint(0, Pose(Eigen::Vector2d(0, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(1, 0), 0);
+            secondPoint = new PathPoint(1, Pose(Eigen::Vector2d(1, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(1, 0), 0);
+        }
+
+        virtual void TearDown() {
+            delete firstPoint;
+            delete secondPoint;
+        }
+};
+
+class yChangeConstantVelModule : public ::testing::Test {
+    protected:
+        PathPoint* firstPoint;
+        PathPoint* secondPoint;
+
+        virtual void SetUp() {
+            // Creates two points that only have a positive change in y with constant velocity
+            firstPoint = new PathPoint(0, Pose(Eigen::Vector2d(0, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(0, 1), 0);
+            secondPoint = new PathPoint(1, Pose(Eigen::Vector2d(0, 1), Eigen::Rotation2Dd(0)), Eigen::Vector2d(0, 1), 0);
+        }
+
+        virtual void TearDown() {
+            delete firstPoint;
+            delete secondPoint;
+        }
+};
+
+class xyChangeConstantVelModule : public ::testing::Test {
+    protected:
+        PathPoint* firstPoint;
+        PathPoint* secondPoint;
+
+        virtual void SetUp() {
+            // Creates two points that have a positive change in x and y with constant velocity
+            firstPoint = new PathPoint(0, Pose(Eigen::Vector2d(0, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(1, 1), 0);
+            secondPoint = new PathPoint(1, Pose(Eigen::Vector2d(1, 1), Eigen::Rotation2Dd(0)), Eigen::Vector2d(1, 1), 0);
+        }
+
+        virtual void TearDown() {
+            delete firstPoint;
+            delete secondPoint;
+        }
+};
+
+class noChangeModule : public ::testing::Test {
+    protected:
+        PathPoint* firstPoint;
+        PathPoint* secondPoint;
+
+        virtual void SetUp() {
+            // Creates two points that do not change
+            firstPoint = new PathPoint(0, Pose(Eigen::Vector2d(0, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(0, 0), 0);
+            secondPoint = new PathPoint(1, Pose(Eigen::Vector2d(0, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(0, 0), 0);
+        }
+
+        virtual void TearDown() {
+            delete firstPoint;
+            delete secondPoint;
+        }
+};
+
+TEST_F(xChangeConstantVelModule, xChangeConstantVel) {
```

Comment:
I think this probably could have been done without test fixtures by doing something like:

TEST(test_thing) {
    PathPoint * firstPoint = new Pathpoint(...);
    PathPoint * secondPoint = new PathPoint(...);

    EXPECT(...);
}

Reply:
Test fixtures are really supposed to be used for more secondary things that are not related to the test, IE setting up a database with fake info that is used in a whole bunch of tests

Ideally a TEST statement should include everything related to what is being tested, the inputs, then the action that is taken to create outputs, and then the expected outputs